src/environment.go

```diff
@@ -77,24 +85,24 @@ type environmentInfo interface {
 	isWsl() bool
 	stackCount() int
 	getTerminalWidth() (int, error)
+	cache() cache
+	close()
 }
 
 type commandCache struct {
-	commands map[string]string
-	lock     sync.RWMutex
+	commands *concurrentMap
 }
 
 func (c *commandCache) set(command, path string) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	c.commands[command] = path
+	c.commands.set(command, path)
 }
 
 func (c *commandCache) get(command string) (string, bool) {
-	c.lock.RLock()
-	defer c.lock.RUnlock()
-	if cmd, ok := c.commands[command]; ok {
-		command = cmd
+	cmd, found := c.commands.get(command)
+	if !found {
+		return "", false
+	}
+	if command, ok := cmd.(string); ok {
```

Comment:
remove this and simply use `return cmd.(string)`

Reply:
I did the same, returning the value with a bool to tell if found or not. It's easier to use after for the caller.